### PR TITLE
To `polkadot-v0.9.38`

### PR DESCRIPTION
**Pull request by jiguantong**
- [x] frontier
- [x] moonbeam
- [x] cumulus
- [x] darwinia-messages-substrate
- [x] remove substrate&cumulus patch
- [x] fix compilation
- [x] replace prepare branch
- [ ] merge after darwinia2.0 is online

### CODE REVIEW

**Review by AurevoirXavier — CHANGES_REQUESTED** [review]
We should keep the assertion patch.

**Comment by jiguantong** [comment]
Cumulus patch branch: https://github.com/paritytech/cumulus/compare/polkadot-v0.9.38...darwinia-network:cumulus:remove-assertion-v0.9.38

File: Cargo.toml [hunk 1]
```diff
@@ -71,307 +71,144 @@ pangolin-runtime                  = { path = "runtime/pangolin" }
...
+cumulus-pallet-xcm                    = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-pallet-xcmp-queue             = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-primitives-core               = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-primitives-parachain-inherent = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-primitives-timestamp          = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-primitives-utility            = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+cumulus-relay-chain-interface         = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
+parachain-info                        = { git = "https://github.com/darwinia-network/cumulus", branch = "remove-assertion-v0.9.38" }
```
Comment (AurevoirXavier) [inline Cargo.toml:214]: Why don't we need the `default-features = false` here?

File: Cargo.toml [hunk 2]
```diff
@@ -71,307 +71,144 @@ pangolin-runtime                  = { path = "runtime/pangolin" }
...
+sp-io                                      = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-keyring                                 = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-keystore                                = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-offchain                                = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-runtime                                 = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-session                                 = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-std                                     = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
+sp-timestamp                               = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
```
Comment (AurevoirXavier) [inline Cargo.toml:187]: Why don't we need the `default-features = false` here?

File: Cargo.toml [hunk 3]
```diff
@@ -71,307 +71,144 @@ pangolin-runtime                  = { path = "runtime/pangolin" }
...
-substrate-build-script-utils                  = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-substrate-frame-rpc-system                    = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-substrate-prometheus-endpoint                 = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-substrate-rpc-client                          = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-substrate-state-trie-migration-rpc            = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-substrate-wasm-builder                        = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
-try-runtime-cli                               = { git = "https://github.com/darwinia-network/substrate", branch = "polkadot-v0.9.37" }
+frame-benchmarking                         = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.38" }
```
Comment (AurevoirXavier) [inline Cargo.toml:123]: I think we could remove all the `default-features` in the patch section.

File: Cargo.toml [hunk 4]
```diff
@@ -71,307 +71,144 @@ pangolin-runtime                  = { path = "runtime/pangolin" }
...
+fc-db                          = { git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fc-mapping-sync                = { git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fc-rpc                         = { git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fc-rpc-core                    = { git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fp-ethereum                    = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fp-evm                         = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fp-rpc                         = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fp-self-contained              = { default-features = false, git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
+fp-storage                     = { git = "https://github.com/darwinia-network/frontier", branch = "polkadot-v0.9.38" }
```
Comment (AurevoirXavier) [inline Cargo.toml:97]: default features? @boundless-forest

File: Cargo.toml [hunk 5]
```diff
@@ -29,24 +29,24 @@ sha3              = { version = "0.10" }
...
+cumulus-pallet-aura-ext               = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-pallet-dmp-queue              = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-pallet-parachain-system       = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-pallet-session-benchmarking   = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-pallet-xcm                    = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-pallet-xcmp-queue             = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-primitives-core               = { default-features = false, git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.38" }
+cumulus-primitives-parachain-inherent = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.38" }
```
Comment (AurevoirXavier) [inline Cargo.toml:45]: default features?

**Pull request merged**